relayer-cli/src/commands/keys/add.rs

```diff
@@ -11,28 +14,48 @@ use crate::error::Kind;
 #[derive(Clone, Command, Debug, Options)]
 pub struct KeysAddCmd {
     #[options(free, help = "identifier of the chain")]
-    chain_id: ChainId,
+    chain_id: Option<ChainId>,
+
+    #[options(
+        short = "f",
+        help = "the path to the key file (conflicts with --mnemonic)"
+    )]
+    file: Option<PathBuf>,
 
-    #[options(free, help = "the key path and filename")]
-    file: Option<String>,
+    #[options(short = "m", help = "the BIP-49 mnemonic (conflicts with --file)")]
+    mnemonic: Option<String>,
 }
 
 impl KeysAddCmd {
-    fn validate_options(&self, config: &Config) -> Result<KeysAddOptions, String> {
+    fn validate_options(&self, config: &Config) -> Result<KeysAddOptions, BoxError> {
+        let chain_id = self
+            .chain_id
+            .as_ref()
+            .ok_or("missing required chain identifier option")?;
+
         let chain_config = config
-            .find_chain(&self.chain_id)
-            .ok_or_else(|| format!("chain '{}' not found in configuration file", self.chain_id))?;
-
-        let key_filename = self
-            .file
-            .clone()
-            .ok_or_else(|| "missing signer key file".to_string())?;
-
-        Ok(KeysAddOptions {
-            name: chain_config.key_name.clone(),
-            file: key_filename,
-            chain_config: chain_config.clone(),
-        })
+            .find_chain(chain_id)
+            .ok_or_else(|| format!("chain '{}' not found in configuration file", chain_id))?;
+
+        if self.file.is_some() && self.mnemonic.is_some() {
+            return Err("the --file and --mnemonic options are mutually exclusive".into());
```

Comment:
should we use another command for this ? I was using `keys restore` for mnemonics and `keys add` for the file.

Reply:
Will do that!